File: ios/CustomMethods.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

// ALL MODULES/METHODS TO BE EXPOSED TO RN ARE MENTIONED IN "Objective-C" ".m" files
// THIS IS ONLY FOR GETTING DATA TO IOS FROM RN

// Declare the CustomMethods class as a React Native module
// The RCT_EXTERN_MODULE macro exposes Objective-C classes as React Native modules
// CustomMethods is the module name, and NSObject is the superclass
@interface RCT_EXTERN_MODULE(CustomMethods, NSObject)

// Expose the MyMethod method to React Native
// RCT_EXTERN_METHOD macro exposes Objective-C methods as React Native methods
RCT_EXTERN_METHOD(myMethod:(NSString *)param)

@end

File: ios/CustomMethods.swift
//
//  CustomMethods.swift

// WRITE ALL CODES LOGICS HERE THAT IS TO BE USED IN RN

//  the Swift code is structured to define a React Native module with a method that can be called from JavaScript code. The @objc annotations and inheritance from NSObject ensure interoperability with Objective-C, which is necessary for integration with React Native.

import Foundation

@objc(CustomMethods) class CustomMethods: NSObject{
   
  @objc static func requiresMainQueueSetup() -> Bool {
      return true // Set to true if setup on the main thread is required
  } //true(when sending some constants to ios/making UI using UIKit in ios)/false run this module on main thread b4 any JS code executes
  
  @objc public func myMethod(_ param:String){
    print(param)
    
    RNEventEmitter.emitter.sendEvent(withName: "onReady", body: [param]) // send to RN from IOS "onReady" type
  }
}
